File: designer-pdf-viewer/designer-pdf-viewer.go
package designerpdfviewer

import "strings"

func designerPdfViewer(letterHeights []int32, word string) int32 {
	alphabet := "abcdefghijklmnopqrstuvwxyz"

	letterHeightMax := int32(0)
	for _, letter := range word {
		alphabetIdx := strings.Index(alphabet, string(letter))
		letterHeight := letterHeights[alphabetIdx]
		if letterHeight > letterHeightMax {
			letterHeightMax = letterHeight
		}
	}

	return letterHeightMax * int32(len(word))
}

File: designer-pdf-viewer/DesignerPdfViewer.java
public class DesignerPdfViewer {

  public static int designerPdfViewer(List<Integer> letterHeights, String word) {
    String alphabet = "abcdefghijklmnopqrstuvwxyz";

    int letterHeightMax = 0;
    for (int i = 0; i < word.length(); i++) {
      char letter = word.charAt(i);
      int alphabetIdx = alphabet.indexOf(letter);
      int letterHeight = letterHeights.get(alphabetIdx);
      letterHeightMax = Math.max(letterHeightMax, letterHeight);
    }

    return letterHeightMax * word.length();
  }

}
